class-tour.php

```diff
@@ -44,10 +44,28 @@ public static function enqueue_scripts() {
 		}
 		$once = true;
 
-		$tours = apply_filters( 'tour_list', array() );
-		if ( empty( $tours ) ) {
+		$_tours = apply_filters( 'tour_list', array() );
+		if ( empty( $_tours ) ) {
 			return;
 		}
+		$tours = array_filter(
+			$_tours,
+			function ( $tour ) {
+				$restrict_url = $tour[0]['tour_restrict_url'];
+				$url_pattern  = '#/wp-admin/post\.php\?post=#';
+				if ( empty( $tour[0]['tour_restrict_url'] ) || preg_match( $url_pattern, $_SERVER['REQUEST_URI'] ) ) {
+					return true;
+				}
+				if ( ! preg_match( '#^/.*$#', $restrict_url ) ) {
+					$restrict_url = preg_replace( '#^https?://(www\.)?[^/]+#', '', $restrict_url );
```

Comment:
So if the restrict URl starts with a `/` it will match against the whole URL but if not you potentially remove URL parts? What if we just extracted a that via `parse_url( $restrict_url, PHP_URL_PATH )`?

Reply:
The regex used before this update didn't seem to cover some URL formats and I'm trying to cover for all possible ways a URL can be written, when creating a tour and I can already identify some cases that are not covered by the selines above.